### ISSUE TITLE
Support "required" packages like `matrices`

### ISSUE THREAD

**Issue by fingolfin** [fingolfin/ReleaseTools#24]
Currently, these can and will yield an error, because we use `SetPackagePath` -- but `SetPackagePath` generates an error if the package is already loaded (which is of course always the case for required packages), but a different copy of it than the one we are asking for with `SetPackagePath`.

(This also affects autoloading packages like io,  when manully running `makeoc.g`, except that there one can work around it by invoke gap with `-A`).

The reason we use `SetPackagePath` in the first place is to ensure that the documentation is really built for the correct package. But I am not quite sure if we *really* need that (mainly because I forgot all the original reasons for it). One reason I do remember is `AutodDoc`, which currently does this in its `AutoDoc()` function:
```
        package_info := PackageInfo( pkg )[ 1 ];
        pkg_dir := DirectoriesPackageLibrary( pkg, "" )[1];
```
So it always uses the first PackageInfo. But it should be possible to improve that in AutoDoc. E.g. we could make it possible to pass the package info record and the package dir to it as a parameter. And/or it could search through the list of packageinfos, and "prefer" any whose install path is the local directory

**Comment by fingolfin** [comment]
We now released AutoDoc 2016.01.21, which makes it unnecessary to run `SetPackagePath`. Unfortunately, it is still up to the packages to update their `makdoc.g` to use that correctly!

So the plan now is to wait until a version of GAP ships with that AutoDoc version (or a newer one). Once that happened, we can tell package authors to update their `makedoc.g` suitably. 

@markuspf Until then, we may have to use a modified version of `release` for the matrices package...

**Comment by fingolfin** [comment]
GAP 4.8.2 has been released, with AutoDoc 2016.02.16. So it's time to start pull requests for packages to use the new, better `makedoc.g` style.

**Comment by fingolfin** [comment]
This has been resolved by de6281ad8cac8b3fea7f197b07e03c788da4cc6b
